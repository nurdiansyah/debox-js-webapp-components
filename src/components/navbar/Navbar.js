// @flow

import React from 'react'
import {propsClassNames} from '../utils/classnamesUtils'

export type NavbarProps = {
  children?: React$Node,
  inverse?: boolean,
  placement?: 'full-width' | 'fixed-top' | 'fixed-bottom' | 'sticky-top',
  color?: string,
  role?: string,
  tag?: React$ElementType,
  withContainer?: ?boolean,
  classBlock?: Object,
  className?: string,
  classNames?: Function,
  toggleable?: string | boolean
}

const defaultProps = {
  tag: 'nav',
  toggleable: false,
  light: true,
  placement: 'full-width'
}

const Navbar = (props: NavbarProps) => {
  const {
    toggleable = defaultProps.toggleable,
    className: _className,
    classNames: _classNames,
    inverse,
    placement,
    color,
    tag: Tag = defaultProps.tag,
    role: _role,
    children,
    withContainer = true,
    ..._attributes
  } = props
  const attributes: Object = _attributes

  if (_role || Tag !== 'nav') {
    attributes.role = _role || 'navigation'
  }

  const classNames = propsClassNames({classNames: _classNames})
  let className = _className ? `${_className} ` : ''
  className += classNames(null, {
    toggleable,
    placement: defaultProps.placement ? placement : false,
    color,
    inverse
  })

  return (
    <Tag {...attributes} className={className}>
      {withContainer ? <div className={classNames('container')}>{children}</div> : children}
    </Tag>
  )
}

export default Navbar
