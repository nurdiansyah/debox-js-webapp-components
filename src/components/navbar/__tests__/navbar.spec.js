import React from 'react'

import Navbar from '../Navbar'
import NavbarBrand from '../Brand'
import NavbarToggle from '../Toggler'

describe('<Navbar/>', () => {
  it('should render Navbar', () => {
    const element = shallow(<Navbar />).getElement()
    expect(element).toMatchSnapshot()
    expect(element.type).toEqual('nav')
    expect(element.props.className).toMatch(/\bnavbar\b/)
    expect(element.props.role).toBeUndefined()
  })

  it('should add "navigation" role when not using a `<nav/>`', () => {
    const element = shallow(<Navbar tag="div" />).getElement()
    expect(element.type).toEqual('div')
    expect(element.props.role).toEqual('navigation')
  })

  it('Should add placement fixedTop variation class', () => {
    const element = shallow(<Navbar placement="fixed-top" />).getElement()
    expect(element.props.className).toMatch(/.*\bnavbar\b.*\bfixed-top\b.*/)
  })

  it('should add header with brand', () => {
    const dom = render(
      <Navbar>
        <NavbarBrand>Brand</NavbarBrand>
      </Navbar>
    )
    const brandDom = dom[0].children[0]
    expect(dom).toMatchSnapshot()
    expect(brandDom).not.toBeNull()
    expect(brandDom.name).toEqual('a')
    expect(brandDom.children[0].data).toEqual('Brand')
  })

  it('should render NavbarToggler', () => {
    const wrapper = shallow(<NavbarToggle />)

    const toggleClassName = wrapper.props().className
    expect(toggleClassName).toEqual('navbar-toggler')
  })

  it('should render NavbarToggler with position right', () => {
    const wrapper = shallow(<NavbarToggle position="right" />)
    const toggleClassName = wrapper.props().className
    expect(toggleClassName).toMatch(/.*\bnavbar-toggler-right\b.*/)
  })
})
