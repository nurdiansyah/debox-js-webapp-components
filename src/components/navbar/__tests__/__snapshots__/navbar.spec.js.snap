// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> should add header with brand 1`] = `
<nav
  class=""
>
  <a
    class="brand"
  >
    Brand
  </a>
</nav>
`;

exports[`<Navbar/> should render Navbar 1`] = `
<nav
  className=""
/>
`;
